src/store/timeLayerModal/index.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  createSlice,
+  PayloadAction
+} from '@reduxjs/toolkit';
+
+export interface TimeLayerModalState {
+  layerId: string | null;
+}
+
+const initialState: TimeLayerModalState = {
+  layerId: null
+};
+
+export const slice = createSlice({
+  name: 'timeLayerModal',
+  initialState,
+  reducers: {
+    openTimeLayerModal(state, action: PayloadAction<string>) {
+      state.layerId = action.payload;
+    },
```

Comment:
Very minor and open for discussion, but IMHO it would be clearer to split this into two functions, something like:

1. `setLayerId(layerId: string)`
2. `setModalVisible(visible: boolean)`

Reply:
Thanks, I split the function into `setLayerId` and `setModalVisible` for clarity.